### PR TITLE
AP_OpticalFlow: allow calibrator to be compiled out

### PR DESCRIPTION
**Pull request by peterbarker**
Not needed on AP_Periph

Also correct for when logging is not enabled - also needed for AP_Periph for the most part

### CODE REVIEW

**Comment by peterbarker** [comment]
```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *      *           *       *                 *      *      *
HerePro             *                                                                     
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                *      *           *       *                 *      *      *
MatekF405                      *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot             *                  *       *                 *      *      *
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      *      *           *       *                 *      *      *
skyviper-v2450                                    *                                       
```

File: libraries/AP_OpticalFlow/AP_OpticalFlow.cpp [hunk 1]
```diff
@@ -257,12 +260,14 @@ void AP_OpticalFlow::stop_calibration()
...
     }
 }
+#endif
 
 void AP_OpticalFlow::update_state(const OpticalFlow_state &state)
 {
     _state = state;
     _last_update_ms = AP_HAL::millis();
 
+#if HAL_LOGGING_ENABLED
```
Comment (tridge) [inline libraries/AP_OpticalFlow/AP_OpticalFlow.cpp:270]: wrong line?

File: libraries/AP_OpticalFlow/AP_OpticalFlow.cpp [hunk 2]
```diff
@@ -295,7 +302,7 @@ void AP_OpticalFlow::Log_Write_Optflow()
     };
     logger->WriteBlock(&pkt, sizeof(pkt));
 }
-
+#endif
```
Comment (tridge) [inline libraries/AP_OpticalFlow/AP_OpticalFlow.cpp:305]: endif marker text

File: libraries/AP_OpticalFlow/AP_OpticalFlow_Calibrator.cpp [hunk 3]
```diff
@@ -323,3 +333,6 @@ void AP_OpticalFlow_Calibrator::log_sample(uint8_t axis, uint8_t sample_num, flo
         (double)body_rate,
         (double)los_pred);
 }
+#endif
```
Comment (tridge) [inline libraries/AP_OpticalFlow/AP_OpticalFlow_Calibrator.cpp:336]: endif marker text

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_OpticalFlow/AP_OpticalFlow.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_OpticalFlow/AP_OpticalFlow.cpp]: Fixed

File: libraries/AP_OpticalFlow/AP_OpticalFlow_Calibrator.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/AP_OpticalFlow/AP_OpticalFlow_Calibrator.cpp]: Fixed

**Pull request merged**